core/src/main/java/org/elasticsearch/index/shard/IllegalIndexShardStateException.java

```diff
@@ -19,26 +19,26 @@
 
 package org.elasticsearch.index.shard;
 
+import org.elasticsearch.ResourceNotFoundException;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.rest.RestStatus;
 
 import java.io.IOException;
 
 /**
  *
  */
-public class IllegalIndexShardStateException extends IndexShardException {
+public class IllegalIndexShardStateException extends ResourceNotFoundException {
```

Comment:
If we add a ResourceNotAvailableException, I think it will be a better base class here. Shard is there, but not ready to make the operation.